lib/apiservers/engine/backends/container.go

```diff
@@ -1405,41 +1406,43 @@ func validateCreateConfig(config *types.ContainerCreateConfig) error {
 	}
 
 	// validate port bindings
-	if config.HostConfig != nil {
-		var ips []string
-		if addrs, err := publicIPv4Addrs(); err != nil {
-			log.Warnf("could not get address for public interface: %s", err)
-		} else {
-			ips = make([]string, len(addrs))
-			for i := range addrs {
-				ips[i] = addrs[i].IP.String()
-			}
+	var ips []string
+	if addrs, err := publicIPv4Addrs(); err != nil {
+		log.Warnf("could not get address for public interface: %s", err)
+	} else {
+		ips = make([]string, len(addrs))
+		for i := range addrs {
+			ips[i] = addrs[i].IP.String()
 		}
+	}
 
-		for _, pbs := range config.HostConfig.PortBindings {
-			for _, pb := range pbs {
-				if pb.HostIP != "" && pb.HostIP != "0.0.0.0" {
-					// check if specified host ip equals any of the addresses on the "client" interface
-					found := false
-					for _, i := range ips {
-						if i == pb.HostIP {
-							found = true
-							break
-						}
-					}
-					if !found {
-						return InternalServerError("host IP for port bindings is only supported for 0.0.0.0 and the public interface IP address")
+	for _, pbs := range config.HostConfig.PortBindings {
+		for _, pb := range pbs {
+			if pb.HostIP != "" && pb.HostIP != "0.0.0.0" {
+				// check if specified host ip equals any of the addresses on the "client" interface
+				found := false
+				for _, i := range ips {
+					if i == pb.HostIP {
+						found = true
+						break
 					}
 				}
-
-				start, end, _ := nat.ParsePortRangeToInt(pb.HostPort)
-				if start != end {
-					return InternalServerError("host port ranges are not supported for port bindings")
+				if !found {
+					return InternalServerError("host IP for port bindings is only supported for 0.0.0.0 and the public interface IP address")
 				}
 			}
+
+			start, end, _ := nat.ParsePortRangeToInt(pb.HostPort)
+			if start != end {
+				return InternalServerError("host port ranges are not supported for port bindings")
+			}
 		}
 	}
 
+	if config.Config == nil {
```

Comment:
I can do the check at the beginning, but I find that all the subsequent checks for config.Config.XXX are right below the current line. So I thought it might be better to move "if config.Config == nil" and "if config.Config.XXX" closer to each other.